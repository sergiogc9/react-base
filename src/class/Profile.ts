export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
