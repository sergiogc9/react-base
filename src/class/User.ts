export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
