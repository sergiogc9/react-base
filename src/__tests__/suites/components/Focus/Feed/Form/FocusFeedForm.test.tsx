import React from "react";
import { shallow } from "enzyme";

import FocusFeedForm from "@src/components/Focus/Feed/Form/FocusFeedForm";
import { TenantObject, Settings } from "@src/class/Tenant";

const tenantSettings: Settings = {
	categorization_mode: 'flc',
	currency: 'EUR',
	display_influencers: false,
	document_tenant_indices_search: true,
	facebook_url: 'fake url',
	valuation_metric: 'miv'
};

const tenant: TenantObject = {
	id: 'rd-girona-test',
	guid: '00034972-0000-0000-0000-000000000000',
	name: 'rd.girona.test',
	tier_properties: {
		name: 'diamond',
		limit: 50000,
		results: {
			online: true,
			social: true
		}
	},
	print_only: false,
	facebook_linked_ids: [
		'10156337492683158',
		'10160315799795215',
		'206028773584551',
		'611466659195813'
	],
	settings: tenantSettings
};

describe("FocusFeedForm", () => {
	let wrapper: any;
	let onSetSort = jest.fn();
	let onSetPeriod = jest.fn();
	let onRecoverFeed = jest.fn();

	const getComponent = () => <FocusFeedForm
		sort="publication_date:desc"
		feedType="socialmedia"
		searchPeriod="last_month"
		start={0}
		limit={20}
		total={0}
		loadingDocuments={false}
		onSetSort={onSetSort}
		onSetPeriod={onSetPeriod}
		onRecoverFeed={onRecoverFeed}
		recoveryId={null}
		recoveryProgress={0}
		tenant={tenant}
	/>;

	beforeEach(() => {
		onSetSort = jest.fn();
		onSetPeriod = jest.fn();
		onRecoverFeed = jest.fn();
		wrapper = shallow(getComponent());
	});

	it("Component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot without feed type", () => {
		wrapper.setProps({ feedType: null });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with print feed", () => {
		wrapper.setProps({ feedType: "print_percolator" });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with recovery id", () => {
		wrapper.setProps({ recoveryId: "234", recoveryProgress: 75 });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component recover feed button handler", () => {
		wrapper = shallow(getComponent());
		wrapper.find('#focusFeedFormRecoverBtn').at(0).simulate('click');
		expect(onRecoverFeed).toHaveBeenCalledTimes(1);
	});

	it("Component sort select handler", () => {
		wrapper = shallow(getComponent());
		wrapper.find('#focusFeedSortBySelectField').at(0).props().onChange('sort-value');
		expect(onSetSort).toHaveBeenCalledTimes(1);
		expect(onSetSort).toHaveBeenCalledWith("sort-value");
	});

	it("Component snapshot with custom period", () => {
		wrapper.setProps({ searchPeriod: "custom", beginDate: new Date(0), endDate: new Date() });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Component snapshot with date range animation in action", () => {
		wrapper.setState({ dateRangeAnimating: true });
		wrapper.setProps({ searchPeriod: "last_month", beginDate: new Date(0), endDate: new Date() });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Component transition exited", () => {
		wrapper = shallow(getComponent());
		wrapper.setState({ dateRangeAnimating: true });
		wrapper.find('#focusFeedDateSelectorsTransition').at(0).props().onExited();
		expect(wrapper.state().dateRangeAnimating).toBe(false);
	});

	it("Component set period change event", () => {
		wrapper = shallow(getComponent());
		wrapper.find('#focusFeedFormPeriodSelect').at(0).props().onChange('last_3_months');
		expect(onSetPeriod).toHaveBeenCalledTimes(1);
		expect(onSetPeriod).toHaveBeenCalledWith('last_3_months');
	});

	it("Component set period change event from custom to not custom period", () => {
		wrapper = shallow(getComponent());
		wrapper.setProps({ searchPeriod: "custom", beginDate: new Date(0), endDate: new Date() });
		wrapper.find('#focusFeedFormPeriodSelect').at(0).props().onChange('last_months');
		expect(wrapper.state().dateRangeAnimating).toBe(true);
	});

	it("Component snapshot with total value", () => {
		wrapper.setProps({ total: 2000 });
		expect(wrapper.html()).toMatchSnapshot();
	});
	it("Component snapshot with total value bigger than 5000", () => {
		wrapper.setProps({ total: 10000 });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with total value smaller than limit", () => {
		wrapper.setProps({ total: 5 });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test audience option sort", () => {
		expect(wrapper.find("#focusFeedSortBySelectField").at(0).props().menuItems[0].value).toEqual('reach:desc');
	});
});
