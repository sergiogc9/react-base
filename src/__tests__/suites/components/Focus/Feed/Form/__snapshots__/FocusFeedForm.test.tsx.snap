// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusFeedForm Component snapshot 1`] = `"<div id=\\"focusFeedForm\\"><div><div><span class=\\"focus-feed-results-preview-text\\">feed.labels.results</span></div><button id=\\"focusFeedFormRecoverBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>feed.form.recovery.button</button></div><div class=\\"flex-end\\"><div id=\\"focusFeedSortBySelectField-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedSortBySelectField\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">results.sort.value</label><div id=\\"focusFeedSortBySelectField-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">results.sort.publication_date</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSortBySelectField\\" value=\\"publication_date:desc\\"/></div></div></div><div id=\\"focusFeedFormPeriodContainer\\"><div id=\\"focusFeedDateSelectors\\"></div><div id=\\"focusFeedFormPeriodSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedFormPeriodSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">filters.period.value</label><div id=\\"focusFeedFormPeriodSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">filters.period.last_month</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedFormPeriodSelect\\" value=\\"last_month\\"/></div></div></div></div></div></div>"`;

exports[`FocusFeedForm Component snapshot with custom period 1`] = `
"<div id=\\"focusFeedForm\\">
  <div>
    <div>
      <span className=\\"focus-feed-results-preview-text\\">
        feed.labels.results
      </span>
    </div>
    <withInk(withTooltip(Button)) id=\\"focusFeedFormRecoverBtn\\" flat={true} onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
      feed.form.recovery.button
    </withInk(withTooltip(Button))>
  </div>
  <div className=\\"flex-end\\">
    <SelectField id=\\"focusFeedSortBySelectField\\" label=\\"results.sort.value\\" repositionOnResize={true} menuItems={{...}} onChange={[Function: onChange]} defaultValue=\\"publication_date:desc\\" value=\\"publication_date:desc\\" anchor={{...}} fixedTo={{...}} position=\\"tl\\" itemLabel=\\"label\\" itemValue=\\"value\\" itemProps=\\"getProps\\" dropdownIcon={{...}} lineDirection=\\"left\\" defaultVisible={false} keyboardMatchingTimeout={1000} transitionName=\\"md-drop\\" transitionTime={300} repositionOnScroll={true} />
    <div id=\\"focusFeedFormPeriodContainer\\">
      <div id=\\"focusFeedDateSelectors\\">
        <CSSTransition id=\\"focusFeedDateSelectorsTransition\\" classNames=\\"focus-feed-form-show-date-selectors\\" timeout={500} in={true} onExited={[Function: onExited]}>
          <Connect(SearchDateRange) />
        </CSSTransition>
      </div>
      <SelectField id=\\"focusFeedFormPeriodSelect\\" label=\\"filters.period.value\\" placeholder=\\"filters.period.value\\" menuItems={{...}} defaultValue=\\"custom\\" value=\\"custom\\" position=\\"tr\\" onChange={[Function: onChange]} anchor={{...}} fixedTo={{...}} itemLabel=\\"label\\" itemValue=\\"value\\" itemProps=\\"getProps\\" dropdownIcon={{...}} lineDirection=\\"left\\" defaultVisible={false} keyboardMatchingTimeout={1000} transitionName=\\"md-drop\\" transitionTime={300} repositionOnScroll={true} repositionOnResize={false} />
    </div>
  </div>
</div>"
`;

exports[`FocusFeedForm Component snapshot with date range animation in action 1`] = `
"<div id=\\"focusFeedForm\\">
  <div>
    <div>
      <span className=\\"focus-feed-results-preview-text\\">
        feed.labels.results
      </span>
    </div>
    <withInk(withTooltip(Button)) id=\\"focusFeedFormRecoverBtn\\" flat={true} onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
      feed.form.recovery.button
    </withInk(withTooltip(Button))>
  </div>
  <div className=\\"flex-end\\">
    <SelectField id=\\"focusFeedSortBySelectField\\" label=\\"results.sort.value\\" repositionOnResize={true} menuItems={{...}} onChange={[Function: onChange]} defaultValue=\\"publication_date:desc\\" value=\\"publication_date:desc\\" anchor={{...}} fixedTo={{...}} position=\\"tl\\" itemLabel=\\"label\\" itemValue=\\"value\\" itemProps=\\"getProps\\" dropdownIcon={{...}} lineDirection=\\"left\\" defaultVisible={false} keyboardMatchingTimeout={1000} transitionName=\\"md-drop\\" transitionTime={300} repositionOnScroll={true} />
    <div id=\\"focusFeedFormPeriodContainer\\">
      <div id=\\"focusFeedDateSelectors\\">
        <CSSTransition id=\\"focusFeedDateSelectorsTransition\\" classNames=\\"focus-feed-form-show-date-selectors\\" timeout={500} in={false} onExited={[Function: onExited]}>
          <Connect(SearchDateRange) />
        </CSSTransition>
      </div>
      <SelectField id=\\"focusFeedFormPeriodSelect\\" label=\\"filters.period.value\\" placeholder=\\"filters.period.value\\" menuItems={{...}} defaultValue=\\"last_month\\" value=\\"last_month\\" position=\\"tr\\" onChange={[Function: onChange]} anchor={{...}} fixedTo={{...}} itemLabel=\\"label\\" itemValue=\\"value\\" itemProps=\\"getProps\\" dropdownIcon={{...}} lineDirection=\\"left\\" defaultVisible={false} keyboardMatchingTimeout={1000} transitionName=\\"md-drop\\" transitionTime={300} repositionOnScroll={true} repositionOnResize={false} />
    </div>
  </div>
</div>"
`;

exports[`FocusFeedForm Component snapshot with print feed 1`] = `"<div id=\\"focusFeedForm\\"><div><div><span class=\\"focus-feed-results-preview-text\\">feed.labels.results</span></div><button id=\\"focusFeedFormRecoverBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>feed.form.recovery.button</button></div><div class=\\"flex-end\\"><div id=\\"focusFeedSortBySelectField-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedSortBySelectField\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">results.sort.value</label><div id=\\"focusFeedSortBySelectField-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSortBySelectField\\" value=\\"publication_date:desc\\"/></div></div></div><div id=\\"focusFeedFormPeriodContainer\\"><div id=\\"focusFeedDateSelectors\\"></div><div id=\\"focusFeedFormPeriodSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedFormPeriodSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">filters.period.value</label><div id=\\"focusFeedFormPeriodSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">filters.period.last_month</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedFormPeriodSelect\\" value=\\"last_month\\"/></div></div></div></div></div></div>"`;

exports[`FocusFeedForm Component snapshot with recovery id 1`] = `"<div id=\\"focusFeedForm\\"><div><div><span class=\\"focus-feed-results-preview-text\\">feed.labels.results</span></div><div id=\\"focusFeedFormRecoveryProgress\\"><span>feed.form.recovery.progress</span><div id=\\"focusFeedFormRecoveryProgressLine\\" class=\\"md-progress md-progress--linear\\"><div role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"75\\" style=\\"width:75%\\" class=\\"md-progress--linear-active md-progress--linear-determinate\\"></div></div><span>75%</span></div></div><div class=\\"flex-end\\"><div id=\\"focusFeedSortBySelectField-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedSortBySelectField\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">results.sort.value</label><div id=\\"focusFeedSortBySelectField-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">results.sort.publication_date</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSortBySelectField\\" value=\\"publication_date:desc\\"/></div></div></div><div id=\\"focusFeedFormPeriodContainer\\"><div id=\\"focusFeedDateSelectors\\"></div><div id=\\"focusFeedFormPeriodSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedFormPeriodSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">filters.period.value</label><div id=\\"focusFeedFormPeriodSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">filters.period.last_month</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedFormPeriodSelect\\" value=\\"last_month\\"/></div></div></div></div></div></div>"`;

exports[`FocusFeedForm Component snapshot with total value 1`] = `"<div id=\\"focusFeedForm\\"><div><div><span class=\\"focus-feed-results-preview-text\\">feed.labels.results</span></div><button id=\\"focusFeedFormRecoverBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>feed.form.recovery.button</button></div><div class=\\"flex-end\\"><div id=\\"focusFeedSortBySelectField-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedSortBySelectField\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">results.sort.value</label><div id=\\"focusFeedSortBySelectField-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">results.sort.publication_date</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSortBySelectField\\" value=\\"publication_date:desc\\"/></div></div></div><div id=\\"focusFeedFormPeriodContainer\\"><div id=\\"focusFeedDateSelectors\\"></div><div id=\\"focusFeedFormPeriodSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedFormPeriodSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">filters.period.value</label><div id=\\"focusFeedFormPeriodSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">filters.period.last_month</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedFormPeriodSelect\\" value=\\"last_month\\"/></div></div></div><div class=\\"form-display-items\\"><div class=\\"form-results-display-label\\">results.search.documents.pagination_label</div><div class=\\"form-results-display-content\\">results.search.documents.pagination_info</div></div></div></div></div>"`;

exports[`FocusFeedForm Component snapshot with total value bigger than 5000 1`] = `"<div id=\\"focusFeedForm\\"><div><div><span class=\\"focus-feed-results-preview-text\\">feed.labels.results</span></div><button id=\\"focusFeedFormRecoverBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>feed.form.recovery.button</button></div><div class=\\"flex-end\\"><div id=\\"focusFeedSortBySelectField-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedSortBySelectField\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">results.sort.value</label><div id=\\"focusFeedSortBySelectField-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">results.sort.publication_date</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSortBySelectField\\" value=\\"publication_date:desc\\"/></div></div></div><div id=\\"focusFeedFormPeriodContainer\\"><div id=\\"focusFeedDateSelectors\\"></div><div id=\\"focusFeedFormPeriodSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedFormPeriodSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">filters.period.value</label><div id=\\"focusFeedFormPeriodSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">filters.period.last_month</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedFormPeriodSelect\\" value=\\"last_month\\"/></div></div></div><div class=\\"form-display-items\\"><div class=\\"form-results-display-label\\">results.search.documents.pagination_label</div><div class=\\"form-results-display-content\\">results.search.documents.pagination_info</div></div></div></div></div>"`;

exports[`FocusFeedForm Component snapshot with total value smaller than limit 1`] = `"<div id=\\"focusFeedForm\\"><div><div><span class=\\"focus-feed-results-preview-text\\">feed.labels.results</span></div><button id=\\"focusFeedFormRecoverBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>feed.form.recovery.button</button></div><div class=\\"flex-end\\"><div id=\\"focusFeedSortBySelectField-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedSortBySelectField\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">results.sort.value</label><div id=\\"focusFeedSortBySelectField-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">results.sort.publication_date</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSortBySelectField\\" value=\\"publication_date:desc\\"/></div></div></div><div id=\\"focusFeedFormPeriodContainer\\"><div id=\\"focusFeedDateSelectors\\"></div><div id=\\"focusFeedFormPeriodSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"focusFeedFormPeriodSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">filters.period.value</label><div id=\\"focusFeedFormPeriodSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">filters.period.last_month</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedFormPeriodSelect\\" value=\\"last_month\\"/></div></div></div><div class=\\"form-display-items\\"><div class=\\"form-results-display-label\\">results.search.documents.pagination_label</div><div class=\\"form-results-display-content\\">results.search.documents.pagination_info</div></div></div></div></div>"`;

exports[`FocusFeedForm Component snapshot without feed type 1`] = `null`;
